### PR TITLE
invalid IBAN processing

### PR DESCRIPTION
**Pull request by kangguru**

### CODE REVIEW

File: lib/iban_calculator.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def self.execute(method, options = {})
     client = Savon.client(wsdl: config.url, logger: config.logger)
     client.call(method, message: options).tap do |response|
       status = response.body[:"#{method}_response"][:return][:result]
-      fail(ServiceError, status) unless status == 'passed'
+      fail(ServiceError, status) unless response.body[:"#{method}_response"][:return_code]
```
Comment (namxam) [inline lib/iban_calculator.rb]: Does this really always work? I am not sure, but I think to remember that the response keys not always had the same format "#{method}_response". But I might be wrong.

Comment (kangguru) [inline lib/iban_calculator.rb]: just took the line that was there :)

File: lib/iban_calculator/iban_validator_response.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ def bank
     end
 
     def account_number
-      @account_number ||= raw_response[:account_number]
+      @account_number ||= string_or_default(raw_response[:account_number], nil)
```
Comment (bumi) [inline lib/iban_calculator/iban_validator_response.rb]: für was is denn das string or default? was ist drin, wenn kein string drin ist?

**Comment by bumi** [comment]
looks good to me. may I merge?

**Comment by kangguru** [comment]
may may!!

**Pull request merged**